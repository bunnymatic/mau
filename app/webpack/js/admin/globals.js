/**
 * Here we import things we want to be global (like jQuery plugins or similar classes)
 * So they can be accessed from views in a script tag
 *
 * An easy way to make them global is simply to stick them on window.
 **/

import MauTimeago from "../app/mau_timeago";
import AdminRoleManager from "./admin_role_manager";
import ApplicationEventsDetailsForm from "./application_events_details_form";
import ArtistsAdminIndex from "./artists_admin_index";
import { GraphPerDay, PlainGraph } from "./graph_helpers";
import MarkItDown from "./mark_it_down";
import MauDatatables from "./mau_datatables";
import SlideToggler from "./slide_toggler";
import StudioArranger from "./studio_arranger";

const globals = {
  AdminRoleManager,
  ArtistsAdminIndex,
  ApplicationEventsDetailsForm,
  GraphPerDay,
  MarkItDown,
  MauDatatables,
  MauTimeago,
  PlainGraph,
  SlideToggler,
  StudioArranger,
};

window.MAU = globals;
