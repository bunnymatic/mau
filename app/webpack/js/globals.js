// go with maps
import "core-js/stable";
import "regenerator-runtime/runtime";

import AccordionShowSectionGivenHash from "./app/accordion_show_section_given_hash";
import AccordionShowSectionWithErrors from "./app/accordion_show_section_with_errors";
import ArrangeArtForm from "./app/arrange_art_form";
import ArtPieceForm from "./app/art_piece_form";
import ArtistBioToggler from "./app/artist_bio_toggler";
import ArtistDetailsToggler from "./app/artist_details_toggler";
import ArtistListInfiniteScroller from "./app/artist_list_infinite_scroller";
import BackToTop from "./app/back_to_top";
import MapResizer from "./app/map_resizer";
import MauMap from "./app/mau_map";
import MobileNavigation from "./app/mobile_navigation";
import Navigation from "./app/navigation";
import PasswordStrengthMeter from "./app/password_strength_meter";
import Sampler from "./app/sampler";
import WhatsThisPopup from "./app/whats_this_popup";

const globals = {
  AccordionShowSectionGivenHash,
  AccordionShowSectionWithErrors,
  ArrangeArtForm,
  ArtistBioToggler,
  ArtistDetailsToggler,
  ArtistListInfiniteScroller,
  ArtPieceForm,
  BackToTop,
  MauMap,
  MapResizer,
  MobileNavigation,
  Navigation,
  PasswordStrengthMeter,
  Sampler,
  WhatsThisPopup,
};
window.MAU = globals;
