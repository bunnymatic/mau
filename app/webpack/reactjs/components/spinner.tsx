import React, { FC } from "react";

interface SpinnerProps {}

export const Spinner: FC<SpinnerProps> = () => {
  return (
    <div className="mau-spinner" role="progressbar">
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
    </div>
  );
};
