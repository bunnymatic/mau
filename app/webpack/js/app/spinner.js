import jQuery from "jquery";
import { Spinner as SpinJS } from "spin.js";

const SPINNER_OPTIONS = {
  color: "#5a636a",
  lines: 12,
  length: 30,
  width: 14,
  radius: 40,
  corners: 1,
  rotate: 0,
  direction: 1,
  speed: 1,
  trail: 60,
  shadow: false,
  hwaccel: false,
  className: "spinner",
  zIndex: 2e9,
  top: "250px",
};

class Spinner {
  constructor(opts) {
    this.opts = opts || {};
    this.el = jQuery(this.opts.element || "#spinner")[0];
    this.spinOpts = { ...SPINNER_OPTIONS, ...this.opts };
  }

  spin() {
    if (!this.spinner) {
      this.spinner = new SpinJS(this.spinOpts);
    }
    return this.spinner.spin(this.el);
  }

  stop() {
    if (this.spinner) {
      return this.spinner.stop();
    }
  }
}

export default Spinner;
