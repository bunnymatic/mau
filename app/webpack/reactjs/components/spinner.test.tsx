import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import React from "react";

import { Spinner } from "./spinner";

describe("Spinner", () => {
  it("looks right (snapshot)", () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });
});
