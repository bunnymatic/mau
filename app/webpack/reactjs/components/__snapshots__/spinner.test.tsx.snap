// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner looks right (snapshot) 1`] = `
<div>
  <div
    class="mau-spinner"
    role="progressbar"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
